dali/operators/numba_function/numba_func.cc

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "dali/operators/numba_function/numba_func.h"
+
+namespace dali {
+
+DALI_SCHEMA(experimental__NumbaFunc)
+  .DocStr(R"code(Invokes a compiled Numba function passed as a pointer.
+
+The run function should be a Numba C callback function (annotated with cfunc). This function is run on a 
+per-sample basis and should have the following definition:
+
+.. code-block:: python
+
+    @cfunc(run_fn_sig([out_numba_types], [in_numba_types]), nopython=True)
```

Comment:
Can we somehow validate if `[out_numba_types], [in_numba_types]` matches what `callback_setup_func` returns?
How about validating the input of the operator inside DALI?